smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -200,254 +140,142 @@ def __init__(
-        """the queue the manager monitors for new tasks"""
+        self._task_queue = task_queue
+        """The dispatcher queue the manager monitors for new tasks"""
         self._feature_store: t.Optional[FeatureStore] = (
             config_loader.get_feature_store()
         )
-        """a feature store to retrieve models from"""
-        self._worker = worker
+        """A feature store to retrieve models from"""
+        self._worker = worker_type()
         """The ML Worker implementation"""
         self._comm_channel_type = comm_channel_type
         """The type of communication channel to construct for callbacks"""
         self._device = device
         """Device on which workers need to run"""
-        self._cached_models: dict[str, t.Any] = {}
-        """Dictionary of previously loaded models"""
-
-    def _validate_request(self, request: InferenceRequest) -> bool:
-        """Ensure the request can be processed.
-        :param request: The request to validate
-        :return: True if the request is valid, False otherwise"""
-        if not self._feature_store:
-            if request.model_key:
-                logger.error("Unable to load model by key without feature store")
-                return False
-
-            if request.input_keys:
-                logger.error("Unable to load inputs by key without feature store")
-                return False
 
-            if request.output_keys:
-                logger.error("Unable to persist outputs by key without feature store")
-                return False
-
-        if not request.model_key and not request.raw_model:
-            logger.error("Unable to continue without model bytes or feature store key")
-            return False
-
-        if not request.input_keys and not request.raw_inputs:
-            logger.error("Unable to continue without input bytes or feature store keys")
-            return False
+        self._device_manager: t.Optional[DeviceManager] = None
+        """Object responsible for model caching and device access"""
+        self._perf_timer = PerfTimer(prefix="w_", debug=True, timing_on=True)
+        """Performance timer"""
 
-        if request.callback is None:
-            logger.error("No callback channel provided in request")
-            return False
-
-        return True
+    def _on_start(self) -> None:
+        self._device_manager = DeviceManager([WorkerDevice(self._device)])
 
+    # pylint: disable-next=too-many-statements
     def _on_iteration(self) -> None:
         """Executes calls to the machine learning worker implementation to complete
         the inference pipeline"""
-        logger.debug("executing worker manager pipeline")
 
-        if self._task_queue is None:
-            logger.warning("No queue to check for tasks")
+        pre_batch_time = time.perf_counter()
+        try:
+            batch: InferenceBatch = self._task_queue.get(timeout=0.0001)
+        except Exception:
```

Comment:
Just adding this comment because I thought of it -- would it be better to catch a specific ``queue.Empty`` exception type?  Other exceptions might be / should be logged in a different way?